certificate-test-fixture/src/main/java/io/netty/testcert/RevocationServer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.testcert;
+
+import com.sun.net.httpserver.HttpServer;
+import io.netty.testcert.x509.CertificateList;
+import io.netty.testcert.x509.Signed;
+
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class RevocationServer {
+    private final HttpServer crlServer;
+    private final String crlBaseAddress;
+    private final ConcurrentHashMap<String, CrlInfo> paths;
+
+    RevocationServer() throws Exception {
+        // Use the built-in HttpServer to avoid any circular dependencies.
+        crlServer = HttpServer.create(new InetSocketAddress(0), 0);
+        crlBaseAddress = "http://localhost:" + crlServer.getAddress().getPort();
+        paths = new ConcurrentHashMap<>();
+        crlServer.createContext("/", exchange -> {
+            if ("GET".equals(exchange.getRequestMethod())) {
+                String path = exchange.getRequestURI().getPath();
+                CrlInfo info = paths.get(path);
+                if (info == null) {
+                    exchange.sendResponseHeaders(404, 0);
+                    exchange.close();
+                    return;
+                }
+                byte[] crl = generateCrl(info);
+                exchange.getResponseHeaders().put("Content-Type", Collections.singletonList("application/pkix-crl"));
+                exchange.sendResponseHeaders(200, crl.length);
+                try (OutputStream out = exchange.getResponseBody()) {
+                    out.write(crl);
+                    out.flush();
+                }
+            } else {
+                exchange.sendResponseHeaders(405, 0);
+            }
+            exchange.close();
+        });
+    }
+
+    public void start() {
+        if (Thread.currentThread().isDaemon()) {
+            crlServer.start();
+        } else {
+            // It's important the CRL server creates a daemon thread,
+            // Users of FakeCAs don't expect to close or stop a server.
+            // By using daemon threads, we won't stop test runs from terminating.
```

Comment:
Not sure that's a good idea, tests will end up launching a million CRL servers

Reply:
Only if they need them, though. They're not implicitly launched.
We could, however, make the server a singleton, and dictate unique URLs for each issuer. In that case, it'd be even more important that it runs in a daemon thread.